include/seqan3/alphabet/concept_detail.hpp

```diff
@@ -0,0 +1,220 @@
+}
+
+/*!\brief Implementation of \link std::ostream & operator<<(std::ostream & os, alphabet_type const a) \endlink
+ * that delegates to `a.to_char()`.
+ * \tparam alphabet_type Must provide a `.to_char()` member function.
+ * \param os The output stream you are printing to.
+ * \param a The alphabet letter that you wish to convert to char.
+ * \returns A reference to the output stream.
+ */
+template <typename alphabet_type>
+std::ostream & operator<<(std::ostream & os, alphabet_type const a)
+//!\cond
+    requires requires (alphabet_type a) { { a.to_char() } -> underlying_char_t<alphabet_type>; }
+//!\endcond
+{
+    os << a.to_char();
+    return os;
+}
+
+/*!\brief Implementation of \link rank_type seqan3::to_rank(semi_alphabet_concept const a) \endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide a `.to_rank()` member function.
+ * \param a The alphabet letter that you wish to convert to rank.
+ * \returns The letter's value in the alphabet's rank type (usually a `uint*_t`).
+ */
+template <typename alphabet_type>
+constexpr underlying_rank_t<alphabet_type> to_rank(alphabet_type const a)
+    requires requires (alphabet_type a) { { a.to_rank() } -> underlying_rank_t<alphabet_type>; }
+{
+    return a.to_rank();
+}
+
+/*!\brief Implementation of
+ * \link alphabet_concept && seqan3::assign_char(alphabet_concept && a, char_type const c)\endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide an `.assign_char()` member function.
+ * \param a The alphabet letter that you wish to assign to.
+ * \param c The `char` value you wish to assign.
+ * \returns A reference to the alphabet letter you passed in.
+ */
+template <typename alphabet_type>
+constexpr alphabet_type & assign_char(alphabet_type & a, underlying_char_t<alphabet_type> const c)
+    requires requires (alphabet_type a) { { a.assign_char(char{0}) } -> alphabet_type &; }
+{
+    return a.assign_char(c);
+}
+
+/*!\brief Implementation of
+ * \link alphabet_concept && seqan3::assign_char(alphabet_concept && a, char_type const c)\endlink
+ * that delegates to a member function.
+ * \tparam alphabet_type Must provide an `.assign_char()` member function.
+ * \param a An alphabet letter temporary.
+ * \param c The `char` value you wish to assign.
+ * \returns The assignment result as a temporary.
+ * \details
+ * Use this e.g. to newly create alphabet letters from char:
+ * ~~~{.cpp}
+ * auto l = assign_char(dna5{}, 'G');  // l is of type dna5
+ * ~~~
+ */
+template <typename alphabet_type>
+constexpr alphabet_type && assign_char(alphabet_type && a, underlying_char_t<alphabet_type> const c)
+    requires requires (alphabet_type a) { { a.assign_char(char{0}) } -> alphabet_type &; }
+{
+    return std::move(a.assign_char(c));
+}
+
+/*!\brief Implementation of
+ * \link semi_alphabet_concept && seqan3::assign_rank(semi_alphabet_concept && a, rank_type const i) \endlink
+ * that delegates to a member function.
```

Comment:
Actually, is it important for the user that the function "delegates to a member function" ? (Besides that this is obvious to see in the one line of code)
IMPOV we should inform the user about WHAT is happening in the \brief declaration and maybe HOW it is done in the details. What do you think?